### PR TITLE
[bug][vue][SIMS-#1302] formio datagrid bugs

### PR DESCRIPTION
**Pull request by ann-aot**
1. Bug was caused due to the `deep: true` enabled in the watch for  `props.data`. and it was introduced as some of the form data were using vue `reactive`.
2. updated designation agreement as it was using  `reactive` to use `ref`
3. resolve `viewOnly` reactive issue of the designation agreement  form
image

### CODE REVIEW

**Review by andrewsignori-aot — APPROVED** [review]
Looks good @ann-aot, great finding, looks good 👍

File: sources/packages/web/src/views/institution/designations/DesignationRequest.vue [hunk 1]
```diff
@@ -79,6 +68,27 @@ export default {
...
           } as DesignationLocationsListItem),
       );
+
+      designationModel.value = {
+        institutionName: institutionState.value.legalOperatingName,
+        institutionType: institutionState.value.institutionType,
+        isBCPrivate: institutionState.value.isBCPrivate,
+        viewMode: DesignationFormViewModes.submission,
+        locations: designationModelLocations,
+        dynamicData: {
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/institution/designations/DesignationRequest.vue:78]: Dynamic data is assigned inside and outside the if. why so ?

Comment (ann-aot) [reply sources/packages/web/src/views/institution/designations/DesignationRequest.vue]: nice catch. left over from my test

**Review by dheepak-aot — APPROVED** [review]
Great analysis and debugging @ann-aot 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @ann-aot

**Pull request merged**